subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -585,6 +694,50 @@ static void hawkbit_dump_deployment(struct hawkbit_dep_res *d)
 	LOG_DBG("%s=%s", "md5sum-http", l->md5sum_http.href);
 }
 
+#ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
+int hawkbit_set_config(struct hawkbit_runtime_config *config)
+{
+	if (k_sem_take(&probe_sem, HAWKBIT_SET_SERVER_TIMEOUT) == 0) {
+		if (config->server_addr != NULL) {
+			strncpy(hb_cfg.server_addr, config->server_addr,
+				sizeof(hb_cfg.server_addr));
+			LOG_DBG("configured %s: %s", "hawkbit/server_addr", hb_cfg.server_addr);
+		}
+		if (config->server_port != 0) {
+			snprintf(hb_cfg.server_port, sizeof(hb_cfg.server_port), "%u",
+				 config->server_port);
+			LOG_DBG("configured %s: %s", "hawkbit/server_port", hb_cfg.server_port);
+		}
+#ifndef CONFIG_HAWKBIT_DDI_NO_SECURITY
+		if (config->auth_token != NULL) {
+			strncpy(hb_cfg.ddi_security_token, config->auth_token,
+				sizeof(hb_cfg.ddi_security_token));
+			LOG_DBG("configured %s: %s", "hawkbit/ddi_token",
+				hb_cfg.ddi_security_token);
+		}
+#endif /* CONFIG_HAWKBIT_DDI_NO_SECURITY */
+		settings_save();
+		k_sem_give(&probe_sem);
+	} else {
+		LOG_WRN("failed setting config");
+		return -EAGAIN;
+	}
+
+	return 0;
+}
+#endif /* CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME */
+
+struct hawkbit_runtime_config hawkbit_get_config(void)
+{
+	struct hawkbit_runtime_config config = {
+		.server_addr = HAWKBIT_SERVER,
+		.server_port = HAWKBIT_PORT_INT,
+		.auth_token = HAWKBIT_DDI_SECURITY_TOKEN,
+	};
+
+	return config;
+}
```

Comment:
I reckon it's more efficient to let the caller to pass the struct into this function by reference, instead of making copies

Reply:
Yeah, hopefully it doesnt grow too much